### DISCUSSION TITLE
Svelte: new RFC stuff

### DISCUSSION THREAD

**Discussion by userquin** [unocss/unocss#479] [General]
new rfc stuff (tracking):

```html
<div style:--some-var={someVar} class="...&color=var(--some-var)">
```

https://github.com/sveltejs/rfcs/blob/master/text/0008-style-directives.md
https://svelte.dev/docs#template-syntax-element-directives-style-property

Latest only,  released in 3.46.0, 3.46.1

**Comment by dominikg** [comment]
The example is sketching how you could use the new style: feature in svelte to pass custom css props to icon mask url parameters

**Reply by userquin** [reply]
somthing like this @dominikg ?

![imagen](https://user-images.githubusercontent.com/6311119/149542082-3d2fad95-8740-4ee3-b9ab-24b23a6621b5.png)

**Reply by userquin** [reply]
```ts
    let someVar = 'blue'
    function toggleLogo() {
        logo = !logo
        someVar = someVar ? 'blue' : 'green'
    }
```

**Reply by userquin** [reply]
@dominikg 

we can use `color-[var(--my-var)]` (not working with the svelte logo since there is no `currentColor` on the svg, just look at the styles). To support it on query param we need to apply the rules on the query params and would need to change the internals, maybe with this approach there is no need to change anything:

![imagen](https://user-images.githubusercontent.com/6311119/149553904-33461908-70e5-4008-8a76-13ab2ef75d59.png)

**Comment by dominikg** [comment]
This was just a quick idea I had when I saw the url params for the mask. No problem if it doesn't work out

[the remark this answers is not in the document]
**Reply by chu121su12** [reply]
There's now the tailwind property syntax `[prop:value]` in preset-mini. Although it will lose to inlined `style:prop=value`. Currently it is a class only as I'm not sure how to support this with attributify. Maybe could you suggest some syntax?

[Playground](https://unocss.antfu.me/?html=DwEwlgbgBAxgNgQwM5ILwCIAWBaJMBOApoQHZQBmchAHlAFYCuSALmOQJ7YynOH5RheAWyRcefKAHMEAB2wAOClWpcA9nHQA%2BAFBQooSLEQoMcgIwAWKACNV%2BEBOvZJ%2BBJzNQkmBCFUB3bAAmajgoXT0IyMiAbWsEGABrF1UGEhA1ODsALgBiGAcAXS1gAHpwCB1wg2h4ZDR0cytbe0dnV3dPb18A4NDwqIGbSWw8hygAQlj4pPwUtIzsnMIQGCLNUvLKvWqjOtNsSxs7B34nFzcDzp9-IJCwwcHo7GxU3C6bmHVFhAsAZjWNpAdICKtogA)

[the remark this answers is not in the document]
**Reply by chu121su12** [reply]
The extraction result looks workable. But using `style` would clash with, well `style` attribute.

![image](https://user-images.githubusercontent.com/379924/149611590-9040c501-d3fc-450d-a293-260db3d2af4a.png)

[the remark this answers is not in the document]
**Reply by userquin** [reply]
Thx @chu121su12 it is about using thre var on the url for the icon not for supporting the bindibg style, this is solved by svelte.

On unplugin icons we can import icons providing custom props vía query params on the import statement.

The problem here is that we Would need to extract the rules from the query params and then parse the rules again and we wont to do this.

Right now the current version allow bg or mask query params on icons name.

I thing using color with brackets to add the var is ok.

[the remark this answers is not in the document]
**Reply by vipero07** [reply]
If I understand the RFC what @dominikg is looking for is for something like an unocss custom rule that replaces or appends property values on existing styles.
```js
rules: [
  [/&(.*?)=(.*)$/, ([,property,value]) => ({[property]: value})]
]
```

This would in theory allow the devs to swap / append properties that could be adjusted with inline styles. I don't think it is svelte dependent though as style is a normal html attribute. The new svelte feature is just the ability to shorthand the style attribute.

My thoughts on the feature though is it kinda blends inline styles with atomic css and I'm not entirely sure what the benefit is when you could write custom rules and variants or in this case just use the `text-(color)` class. However if every class is currently matched against every rule it wouldn't be too difficult to implement as the above rule kinda works (though it breaks every other rule since their regex expects to end), albeit this is a bit naive and only functioning for one property.

I think @chu121su12 hit the nail on the head though, [arbitrary properties](https://tailwindcss.com/docs/adding-custom-styles#arbitrary-properties) could do exactly this:
```svelte
<div class="i-logos-svelte-icon [color=--some-var]" style:--some-var={someVar}>
```

[the remark this answers is not in the document]
**Reply by chu121su12** [reply]
To support this properly, this needs support from core to split incoming token into multiple tokens, either from variant or, preferrably, preprocess. This is to handle case like `url?a=b&c=d&e=f` where currently will only be matched once. This should also help with #295 splitting token for each browser vendor/implementation.
